### PR TITLE
[RR-20] move oom to use dry run

### PR DESCRIPTION
**Pull request by sjpotter**
changes from manually doing it to using dry run, would be a performance regression but will probably need it for acl in future anyways.

didn't require any test changes

### CODE REVIEW

File: src/redisraft.c [hunk 1]
```diff
@@ -646,9 +646,32 @@ static void handleRedisCommandAppend(RedisRaftCtx *rr,
...
-    if (cmd_flags & CMD_SPEC_DENYOOM && (RedisModule_GetUsedMemoryRatio() > 1.0)) {
-        RedisModule_ReplyWithError(ctx, "OOM command not allowed when used memory > 'maxmemory'.");
-        return;
+    /* dry run */
+    for (int i = 0; i < cmds->len; i++) {
+        RaftRedisCommand *cmd = cmds->commands[i];
+        size_t cmdlen;
+        const char *cmdstr = RedisModule_StringPtrLen(cmd->argv[0], &cmdlen);
+        /* skip multi */
+        if (i == 0 && cmdlen == 5 && !strncasecmp(cmdstr, "MULTI", 5)) {
```
Comment (tezc) [inline src/redisraft.c:655]: Why do we skip multi? If I'm not mistaken, ACL can reject it? Even it won't reject it, do we need this special check?

File: src/redisraft.c [hunk 2]
```diff
@@ -646,9 +646,32 @@ static void handleRedisCommandAppend(RedisRaftCtx *rr,
...
+            continue;
+        }
+
+        char *rm_call_flags;
+        if (RedisModule_GetUsedMemoryRatio() > 1.0) {
```
Comment (tezc) [inline src/redisraft.c:660]: As we are going to call dry-run unconditionally, we don't need to do this check here. rm_call() does more or less same thing, checks usage ratio + one extra if check for DENY_OOM flag. So, I think we can pass "DCEMv" directly. 

Hopefully, we can improve rm_call() performance later.

**Review by tezc — APPROVED** [review]

**Comment by tezc** [comment]
@sjpotter Maybe as part of this PR, maybe in another PR, it'd be nice if we can add OOM tests with scripts. With or without allow-oom flags etc.

**Comment by sjpotter** [comment]
so we have a PR already for scripts, but depends on redis updates that haven't gone in yet.

**Comment by tezc** [comment]
Okay, we'll merge those tests later with that PR then.

**Comment by fadidahanna** [comment]
Do we still need `CMD_SPEC_DENYOOM`? I guess the answer is no, so maybe remove it too from our code, to keep it cleaner.

**Review by fadidahanna — APPROVED** [review]

File: tests/integration/test_config.py [hunk 3]
```diff
@@ -278,17 +278,6 @@ def test_module_command_flags(cluster):
     dont_intercept = 1 << 4
     denyoom = 1 << 7
```
Comment (fadidahanna) [inline tests/integration/test_config.py:279]: can be removed too. Also in 'hellomodule.c`

**Review by tezc — APPROVED** [review]

**Review by tezc — APPROVED** [review]

**Pull request merged**